### DISCUSSION TITLE
description_hash says it accepts Hash(SHA-256) of a description but then fails with error UNKNOWN: description hash is 64 bytes, must be 32

### DISCUSSION THREAD

**Discussion by DimoDonchev** [lightningnetwork/lnd#8230] [Troubleshooting]
As far as i know sha-256 result is 64 bytes, but i get the error that the description needs to be 32 bytes. Can someone explain what i should do?

here is my code.

`    
    // Generate the invoice request
    LightningOuterClass.Invoice invoiceRequest = LightningOuterClass.Invoice.newBuilder()
            .setValueMsat(100000)  // Invoice value in milli-satoshis (1 satoshi = 1000 milli-satoshis)
            .setDescriptionHash(ByteString.copyFrom("0004f0d7cf6c4a4d311e48ef6637026718a7d249a1073774a74d1c7c2fffce6b", Charset.defaultCharset()))
            .build();

    // Generate the invoice
    AddInvoiceResponse invoiceResponse;
    try {
      invoiceResponse = stub.addInvoice(invoiceRequest);
    } catch (StatusRuntimeException e) {
      // Handle error
      System.out.println("Error generating invoice: " + e.getMessage());
      return;
    }
 `
    
   and i get this error  `Error generating invoice: UNKNOWN: description hash is 64 bytes, must be 32`

**Comment by guggero** [comment]
A hash is 32 bytes (256bit divided by 8 equals 32byte). But what you have is a hex encoded string, which then has 64 characters. What you need to do is to parse the hex string into raw bytes and use that.
